### PR TITLE
Make `with` param support hash

### PR DESCRIPTION
**Pull request by izumin5210**
## Why / What
for https://github.com/wantedly/pb-serializer/pull/34

### CODE REVIEW

**Comment by izumin5210** [comment]
@qnighy can you review it today? :pray:

File: lib/computed_model.rb [hunk 1]
```diff
@@ -311,18 +311,27 @@ def computing_plan(deps)
   # @return [Hash{Symbol=>Array}]
   def self.normalize_dependencies(deps)
```
Comment (qnighy) [inline lib/computed_model.rb:312]: Please update the yard annotation above:

```diff
-  # @param deps [Array<Symbol, Hash>]
+  # @param deps [Array<Symbol, Hash>, Hash]
```

File: lib/computed_model.rb [hunk 2]
```diff
@@ -311,18 +311,27 @@ def computing_plan(deps)
...
+          end
+        else; raise "Invalid dependency: #{elem.inspect}"
         end
-      else; raise "Invalid dependency: #{elem.inspect}"
       end
+    when Hash
+      deps.each do |key, elem|
+        normalized[key] ||= []
+        normalized[key].push(*Array(elem))
+      end
```
Comment (qnighy) [inline lib/computed_model.rb:333]: This is a duplicate of the `Array<Hash>` case. Could you somehow unify the implementation?

**Review by qnighy — COMMENTED** [review]
Tests would be appreciated!

**Comment by izumin5210** [comment]
🙈

File: lib/computed_model.rb (hunk 2 above)
Comment (izumin5210) [reply lib/computed_model.rb]: fixed e86934f

File: lib/computed_model.rb (hunk 1 above)
Comment (izumin5210) [reply lib/computed_model.rb]: fixed e86934f

**Comment by izumin5210** [comment]
> Tests would be appreciated!

added be1f2c2 ✅

**Comment by izumin5210** [comment]
@qnighy re-review please :pray: :bow: :pray:

File: lib/computed_model.rb [hunk 3]
```diff
@@ -307,24 +307,37 @@ def computing_plan(deps)
...
-          v = [v] if v.is_a?(Hash)
-          normalized[k] ||= []
-          normalized[k].push(*Array(v))
+  class << self
+    # @param deps [Array<Symbol, Hash>, Hash]
+    # @return [Hash{Symbol=>Array}]
+    def normalize_dependencies(deps)
+      normalized = {}
+      case deps
+      when Array
```
Comment (qnighy) [inline lib/computed_model.rb:316]: How about doing `deps = Array(deps)`? That would simplify things and supports `deps: Symbol` at the same time.

Comment (izumin5210) [reply lib/computed_model.rb:316]: !!!!!

Comment (izumin5210) [reply lib/computed_model.rb]: fixed 611462f ✅

**Review by qnighy — APPROVED** [review]
Thanks! Be sure to update the changelog.

**Pull request merged**